### PR TITLE
Licensing service tests

### PR DESCRIPTION
**Pull request by jrmccannon**
Wanted to work through adding unit tests to an open source project. This adds tests around ValidateOrganizationsAsync from LicensingService. Any feedback is greatly appreciated.

### CODE REVIEW

File: test/Core.Test/Services/LicensingServiceTests.cs [hunk 1]
```diff
@@ -38,12 +51,215 @@ public LicensingServiceTests()
...
+            {
+                new Organization
+                {
+                    Id = organizationId,
+                    Enabled = true,
+                    LicenseKey = "test"
+                },
+                new Organization
+                {
+                    Id = organizationId,
```
Comment (cscharf) [inline test/Core.Test/Services/LicensingServiceTests.cs:111]: If ensuring the test covers a 2 different orgs with the same key, each org should have it's own `Id` to ensure expected behavior would follow those lines (vs. the logic testing/asserting 2 orgs don't exist with the same `Id`, etc.).

File: test/Core.Test/Services/LicensingServiceTests.cs [hunk 2]
```diff
@@ -38,12 +51,215 @@ public LicensingServiceTests()
...
+            var fileName = $"{_organizationDirectory}/{organizationId}.json";
+
+            Directory.CreateDirectory(_organizationDirectory);
+
+            File.WriteAllText(fileName, JsonConvert.SerializeObject(license));
+
+            return fileName;
+        }
+
+        private static void DeleteLicenseFile(string licenseFile)
```
Comment (cscharf) [inline test/Core.Test/Services/LicensingServiceTests.cs:259]: I would be curious to know if there's a better way to mock or fake file system access for these license files through the license service vs. writing/reading physical files for unit tests.

Also, I don't see test hooks for overall test execution cleanup to check for errant files and delete them.

File: test/Core.Test/Services/LicensingServiceTests.cs [hunk 3]
```diff
@@ -19,14 +30,16 @@ public class LicensingServiceTests
         private readonly IWebHostEnvironment _hostingEnvironment;
         private readonly ILogger<LicensingService> _logger;
 
+        private static readonly string _organizationDirectory = $"{Environment.CurrentDirectory}/organization";
```
Comment (cscharf) [inline test/Core.Test/Services/LicensingServiceTests.cs:33]: For unit testing, should always use a Temp directory for assets directed by the test framework vs. `Environment.CurrentDirectory`, this gives the test framework and any CI runners in the build pipeline to cleanup after test runs if necessary, ensures proper access rights to disk are maintained and that can be overridden by test framework configuration directives, command-line arguments, etc.

File: test/Core.Test/Services/LicensingServiceTests.cs (hunk 2 above)
Comment (jrmccannon) [reply test/Core.Test/Services/LicensingServiceTests.cs:259]: That's fair. I looked into loading in an In-Memory file system, but the ones I found were relying on being able to swap out the FileSystem using System.IO.Abstractions. I didn't really want to refactor the existing code without tests, so this seemed like a good stepping point.

I'll work on a safe way of cleaning up the directory without affecting the other tests that will run at the same time.

**Comment by cscharf** [comment]
Closing due to age and stagnation.

**Pull request closed without merging**